Assets/Scripts/UI/PlayButton.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+using System.Linq;
+using Dungeons;
+using Matchmaking;
+using UnityEngine;
+
+namespace UI
+{
+    public class PlayButton : MonoBehaviour, IButtonPressHandler
```

Comment:
Instead of putting this code directly on the button, can we add it to the button's OnClick event instead? Maybe this code instead goes in a class called 'Matchmaker', which can live on a GameObject at the root of our hierarchy?